spotiplex/main.py

```diff
@@ -63,15 +63,25 @@ def __init__(self):
         if self.lidarr_sync == "true":
             self.sync_lists = self.lidarr_api.get_lidarr_playlists()
         else:
+            # This should be an array of arrays to be run by multiple 'threads':
+            # For example: [["playlist1"],["playlist2"],["playlist3","playlist4"]]
             self.sync_lists = self.config.get("manual_playlists")
-        currentuser = self.plex_service.plex.myPlexAccount().username.lower()
-        if currentuser in self.user_list:
-            self.user_list.remove(currentuser)
+        print(f"Attempting to run for {self.sync_lists}")
+        self.default_user = self.plex_service.plex.myPlexAccount().username
+
+        # If the the user list provided is empty, add the default user from the token
+        if not self.user_list or len(self.user_list) == 0:
```

Comment:
I think this isn't going to work, the reason the initial version is in there like that is because Plex doesn't handle the default user correctly if its manually specified, unless I'm misunderstanding the purpose of this block

Reply:
Hmm, I wonder if the Plex API has changed since you originally wrote this? 🤔 
In my initial testing trying to use this app, I actually couldn't get this to run _at all_ with the original  system in place.

I had used it with `plex_users = "<plex_username>"` in my `config.toml`, and it would error out with the following issues:
```
Traceback (most recent call last):
  File "/mnt/h/dev/spotify/spotify-to-plex/spotiplex/main.py", line 189, in <module>
    main()
  File "/mnt/h/dev/spotify/spotify-to-plex/spotiplex/main.py", line 184, in main
    spotiplex.run()
  File "/mnt/h/dev/spotify/spotify-to-plex/spotiplex/main.py", line 99, in run
    self.process_for_user(user)
  File "/mnt/h/dev/spotify/spotify-to-plex/spotiplex/main.py", line 73, in process_for_user
    self.plex_service.plex = self.plex_service.plex.switchUser(user)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/justin/.cache/pypoetry/virtualenvs/spotiplex-LFMqgN-A-py3.12/lib/python3.12/site-packages/plexapi/server.py", line 271, in switchUser
    user = user if isinstance(user, MyPlexUser) else self.myPlexAccount().user(user)
                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/justin/.cache/pypoetry/virtualenvs/spotiplex-LFMqgN-A-py3.12/lib/python3.12/site-packages/plexapi/myplex.py", line 669, in user
    raise NotFound(f'Unable to find user {username}')
plexapi.exceptions.NotFound: Unable to find user <username>
```

When I inspected the Spotify API further and debugged the values being returned for the user account's email, username, and display name (which helped validate I had the right username and it wasn't a weird typo), it did not succeed.

In fact, the entire `switchUser` block completely doesn't work for me, and that's why I had to go with the approach of detecting if the user is the default user, and if so to just use the existing already-auth'd `plex` service that's returned 🤔

Reply:
Something about this doesn't work but I don't think its you, I think its somewhere else. Merging and will fix as needed.